src/pages/Search/SearchFilters.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button, Stack } from '@mui/material';
+import SortingTypeField from './SortingTypeField';
+import SortingDirectionField from './SortingDirectionField';
+import { createSearchParams, useNavigate, useSearchParams } from 'react-router-dom';
+import { ROUTES, SEARCH_PARAMS } from '../../const';
+import { removeEmptySearchParams } from '../../utils/utils';
+import DatePickerFilter from './DatePickerFilter';
+import { LocalizationProvider } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+
+const minWidth = 150;
+
+function SearchFilters() {
+  const [searchParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  const search = (key: string, value?: string) => {
+    if (!value) searchParams.delete(key);
+    else {
+      searchParams.set(key, value);
+      navigate({
+        pathname: ROUTES.SEARCH,
+        search: removeEmptySearchParams(searchParams).toString(),
+      });
```

Comment:
obecnie jak zmienisz filtr to przycisk cofania cofa do poprzednich filtrów i każda zmiana filtrów dodaje się do historii. Fix:
```suggestion
      navigate({
        pathname: ROUTES.SEARCH,
        search: removeEmptySearchParams(searchParams).toString(),
      }, {replace: true});
```

Reply:
done